extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/querykit/WindowOperatorQueryKit.java

```diff
@@ -162,6 +207,28 @@ public QueryDefinition makeQueryDefinition(
           );
         }
 
+        log.info("Using row signature [%s] for window stage.", stageRowSignature);
+
+        boolean partitionOperatorExists = false;
+        List<String> currentPartitionColumns = new ArrayList<>();
+        for (OperatorFactory of : operatorList.get(i)) {
+          if (of instanceof NaivePartitioningOperatorFactory) {
+            for (String s : ((NaivePartitioningOperatorFactory) of).getPartitionColumns()) {
+              currentPartitionColumns.add(s);
+              partitionOperatorExists = true;
+            }
+          }
+        }
+
+        if (partitionOperatorExists) {
+          partitionColumnNames = currentPartitionColumns;
+        }
+
+        log.info(
+            "Columns which would be used to define partitioning boundaries for this window stage are [%s]",
+            partitionColumnNames
+        );
```

Comment:
I love this idea!

Can I take it up in a separate future PR though?

For my next PR, I'm working on revamping the logic of `getOperatorListFromQuery()` method to fix scenarios with empty over() clauses. I can either make this refactoring change in that PR, or the one after that.

Thoughts?

Reply:
of course!
I would recommend to separate these as much as possible - many small PRs used to get reviews faster and because of the size them the feedback is usually also much better!